crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -286,10 +286,33 @@ where
 
         // If this is sent from new payload then the parent hash could be in a side chain, and is
         // not necessarily canonical
-        if self.blockchain.header_by_hash(parent_hash).is_some() {
+        if self.blockchain.header_by_hash(parent_hash).is_some() ||
```

Comment:
nit: I would like helper methods for this, can just move `.is_some` calls to a `contains_headers() -> bool` function